### PR TITLE
Search func bug

### PR DESCRIPTION
**Pull request by pjmiles**
working on the search functionality

### CODE REVIEW

File: src/components/form/ImageForm.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import { useState } from "react";
+import axiosInstance from "../api/axios";
+import "./ImageForm.css";
+
+const ImageForm = ({ closeModal }) => {
+  const [userDetails, setUserDetails] = useState({ title: "", image: null });
+  const [message, setMessage] = useState("")
+
+  const handleChange = (e) => {
+    e.preventDefault();
```
Comment (Asylcreek) [inline src/components/form/ImageForm.js:10]: Why prevent default?

Comment (Asylcreek) [reply src/components/form/ImageForm.js]: And what is the default action?

File: src/components/header/Header.js [hunk 2]
```diff
@@ -1,52 +1,54 @@
...
-  },[]);
 
 
+  useEffect(() => {
+    if (!mount) {
+      setMount(true);
+      searchPictures();
+    }
+  }, [searchPictures, mount]);
+
```
Comment (Asylcreek) [inline src/components/header/Header.js:38]: The `searchPictures` declaration should be inside this `useEffect`

File: src/components/search/Search.js [hunk 3]
```diff
@@ -1,21 +1,33 @@
...
+  };
 
-// export default Search;
+  const searchList = () => {
+    return (
+      <>
+        <SearchItems filteredImages={filteredImages} />
+      </>
+    );
+  };
```
Comment (Asylcreek) [inline src/components/search/Search.js:21]: Why?

File: src/components/search/Search.js [hunk 4]
```diff
@@ -1,21 +1,33 @@
...
+
+  return (
+    <>
+      <div>
+        <input type="text" placeholder="Search image" onchange={handleChange} />
+      </div>
+      {searchList()}
```
Comment (Asylcreek) [inline src/components/search/Search.js:28]: Why?

File: src/components/search/SearchItems.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import DisplayImage from '../display/DisplayImage'
+import "./Search.css";
+
+const SearchItems = ({ filteredImage }) => {
+
+const filtered = filteredImage.map(image => <DisplayImage key={image.title} image={image}/>)
```
Comment (Asylcreek) [inline src/components/search/SearchItems.js:6]: Why bring this out of line 10?

**Review by Asylcreek — CHANGES_REQUESTED** [review]

File: src/components/form/ImageForm.js (hunk 1 above)
Comment (pjmiles) [reply src/components/form/ImageForm.js]: I have removed it, its not a form, now corrected in the new ticket.

File: src/components/header/Header.js (hunk 2 above)
Comment (pjmiles) [reply src/components/header/Header.js]: I now used without the useEffect, now works perfectly without re-rendering issue.

File: src/components/search/Search.js (hunk 3 above)
Comment (pjmiles) [reply src/components/search/Search.js]: removed the file already.. I passed in the images and setImages state as props in the component and made use of it where needed in the new ticket.

File: src/components/search/Search.js (hunk 4 above)
Comment (pjmiles) [reply src/components/search/Search.js]: removed

File: src/components/search/SearchItems.js (hunk 5 above)
Comment (pjmiles) [reply src/components/search/SearchItems.js]: removed the file and implemented the search differently.

**Review by Asylcreek — APPROVED** [review]

**Pull request merged**